src/app/frontend/deploy/deploy.controller.js

```diff
@@ -131,4 +143,24 @@ export default class DeployController {
   isPortMappingEmpty_(portMapping) {
     return !!portMapping.port && !!portMapping.targetPort;
   }
+
+  /**
+   * Initializes namespace selector.
+   * @param {!angular.$log} $log
+   * @param {!angular.$resource} $resource
+   * @private
+   */
+  initializeNamespaces_($log, $resource) {
+    /** @type {!angular.Resource<!backendApi.NamespaceList>} */
+    let resource = $resource('/api/namespace');
+
+    resource.get((namespaceList) => {
+      $log.info('Successfully fetched namespace list: ', namespaceList);
+      this.namespaces = namespaceList.namespaces;
+      this.namespace= this.namespaces[0];
+    }, (err) => {
+      $log.error('Error fetching namespace list: ', err);
```

Comment:
It's better to use 'this.log_' here as it is already injected to constructor and assigned.